Realm/RLMObject.mm

```diff
@@ -40,7 +40,11 @@ - (instancetype)init {
 }
 
 - (instancetype)initWithValue:(id)value {
-    return [super initWithValue:value schema:RLMSchema.sharedSchema];
+    RLMSchema *schema = RLMSchema.partialSharedSchema;
+    if (![schema schemaForClassName:self.class.className]) {
```

Comment:
@jpsim it _is_ sufficient to ensure the linked properties classes are already included, because when the classes are registered into the partial shared schema, all of their properties get parsed so the `RLMObjectSchema` can get created, and if a linked property's class were missing, an exception would be thrown.